contracts/EntryPoint.sol

```diff
@@ -253,25 +270,26 @@ contract EntryPoint is StakeManager {
      *      In order to split the running opcodes of the wallet (validateUserOp) from the paymaster's validatePaymasterUserOp,
      *      it should look for the NUMBER opcode at depth=1 (which itself is a banned opcode)
      * @param userOp the user operation to validate.
-     * @param aggregatedSignature: if the wallet is an IAggregatedSignature, then the simulation of the
-     *      signature is done over a single-operation "bundle". The userOp parameter to simulate is
-     *      as it should be during aggregation (with signature removed).
-     *      If the wallet is not aggregatable, then this parameter is empty.
      * @return preOpGas total gas used by validation (including contract creation)
      * @return prefund the amount the wallet had to prefund (zero in case a paymaster pays)
+     * @return aggregator the aggregator used by this userOp. if a non-zero aggregator is returned, the bundler must get its params using
+     *      aggregator.
      */
-    function simulateValidation(UserOperation calldata userOp, bytes calldata aggregatedSignature) external returns (uint256 preOpGas, uint256 prefund) {
-        if (aggregatedSignature.length != 0) {
-            IAggregator aggregator = IAggregator(IAggregatedWallet(userOp.getSender()).getAggregator());
-            UserOperation[] memory arr = new UserOperation[](1);
-            arr[0] = userOp;
-            aggregator.validateSignatures(address(this), arr, aggregatedSignature);
+    function simulateValidation(UserOperation calldata userOp) external returns (uint256 preOpGas, uint256 prefund, address aggregator) {
+
+        aggregator = address(0);
+        bool success = Executor.call(userOp.getSender(), 0, abi.encodeCall(IAggregatedWallet.getAggregator, ()), gasleft());
```

Comment:
This `getAggregator` optional method is barely useful in my opinion. Instead of reading it from the Wallet that may or may not have this method, and also it is an awkward way for the Bundler to discover what aggregator the UserOp uses, this method could accept `aggregator` as a parameter.
The value for the `aggregator` may originally come from the RPC call of `sendUserOperation` as a second parameter.

Reply:
P.S. I think regular solidity `try`/`catch` will be good enough here.